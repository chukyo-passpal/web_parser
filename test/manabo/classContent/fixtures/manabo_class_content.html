<script>
    $(function () {
        // event setup
        $('.a-reload-contents').on('click', function (e) {
            e.preventDefault();
            var pagetype = $(this).attr('page_type');
            glexa.loadClassContents(85131, 1214705, pagetype);
        });
        $('.a-content-open').on('click', function (e) {
            e.preventDefault();
            if ($(this).attr('plugin_key') == 'resource') {
                return false;
            }
            $(this).closest('tr').next('tr').toggle('fast');
            $(this).closest('tr').next('tr').find('.x-content-hide').slideToggle('fast');
        });
        $('.a-fullscreen').on('click', function (e) {
            if (!glexa.isApp()) {
                e.preventDefault();
                glexa.startExam($(this).attr('href'));
            }
        });
        $('.btn-start-locked-content').on('click', function (e) {
            var form = $(this).parent('form:first');
            var is_exam = $(form).find('input[name=is_exam]:first').val();
            var plugin_key = $(form).find('input[name=plugin_key]:first').val();
            if (parseInt(is_exam) == 0 || glexa.isApp() || plugin_key != 'quiz') {
                // 試験モード✖
                // スマホ・タブレット上
                // Quiz
                // 以外は試験モードでの起動をしない
                return true;
            }
            e.preventDefault();
            var content_id = $(form).find('input[name=content_id]:first').val();
            var uniqid = $(form).find('input[name=uniqid]:first').val();
            var password = $(form).find('input[name=password]:first').val();
            var url = '/content/?plugin_key=' + plugin_key + '&content_id=' + content_id + '&uniqid=' + uniqid + '&password=' + password;
            glexa.startExam(url);
        });
        $('.select-load-contents').on('change', function () {
            getContentList(85131, 1214705, $(this).val());
        });
        $('.x-content-hide').hide();
        $('.td-menu').children().hide();
        // first load
        setDirectoryId(1214705);
        glexa.loadClassSyllabus(85131, 1214705);
        if (!glexa.isApp()) {
            $('.exam-message').removeClass('display-none');
        }
    });
    function getContentEditMode() {
        var mode = glexa.sessionStorage.getItem('content_edit_mode');
        return mode;
    }
    function resetContentUI() {
        $('.a-content-menu').css({ opacity: 1 });
        $('.td-content-checkbox').hide();
        $('.td-content-checkbox input').removeAttr('disabled');
        $('.x-content-hide').hide();
        glexa.destroySort({ element: '.tbody-content-sort' });
        glexa.destroyDrag({ dragElement: '.tr-content-drag', dropElement: '.x-content-drop' });
    }
    function setContentEditMode(mode, isForce) {
        var currentMode = getContentEditMode();
        if (!isForce && getContentEditMode() == mode) {
            glexa.loadClassContents(85131, 1214705, 'edit');
            glexa.sessionStorage.removeItem('content_edit_mode');
        } else {
            resetContentUI();
            if (mode) {
                glexa.sessionStorage.setItem('content_edit_mode', mode);
                $('.a-content-menu').css({ opacity: 0.2 });
                $('.a-content-' + mode).css({ opacity: 1 });
                $('.div-edit').hide();
                glexa.inlineEditor('destroy');
                // 機能別処理
                switch (mode) {
                    case 'sort':
                        $('.div-sort-form').show();
                        glexa.setSort({
                            element: '.tbody-content-sort',
                            axis: 'y',
                            onSuccess: saveContentSort
                        });
                        break;
                    case 'move':
                        $('.div-drag-form').show();
                        $('.td-content-checkbox').show();
                        glexa.setDrag({
                            dragElement: '.tr-content-drag',
                            dropElement: '.x-content-drop',
                            helper: helperContent,
                            onDropped: function (e, ui) {
                                contentIds = [];
                                $('.input-content-ids:checked').each(function () {
                                    contentIds.push($(this).val());
                                });
                                if ($(e.target).attr('class_id')) {
                                    params = {
                                        target_class_id: $(e.target).attr('class_id'),
                                        target_directory_id: $(e.target).attr('directory_id'),
                                    }
                                } else {
                                    params = {
                                        target_class_id: 85131,
                                        target_directory_id: $(e.target).attr('directory_id')
                                    }
                                }
                                params.content_ids = contentIds;
                                glexa.ajax({
                                    action: 'teacher_ajax_class_content_list_move_accept',
                                    params: params,
                                    onSuccess: function () {
                                        glexa.loadClassDirectories(85131, 1214705);
                                        glexa.loadClassContents(85131, 1214705, 'edit');
                                    }
                                });
                            }
                        });
                        break;
                    case 'copy':
                    case 'share':
                        $('.div-drag-form').show();
                        $('.div-copy-form').show();
                        $('.td-content-checkbox').show();
                        var dragElement = '.tr-content-drag';
                        if (mode == 'share') {
                            dragElement = '.tr-content-share-drag';
                            $('.td-content-disable-share-checkbox input').attr('disabled', 'disabled');
                            $('.td-content-disable-share-checkbox input').removeAttr('checked');
                            $('.hidden-share').hide();
                        } else {
                            $('.hidden-share').show();
                        }
                        glexa.setDrag({
                            dragElement: dragElement,
                            dropElement: '.x-content-drop',
                            helper: helperContent,
                            onDropped: function (e) {
                                params = {};
                                params.use_pool = $('input[name=use_pool]').prop('checked') ? 1 : 0;
                                params.use_pool_same = $('input[name=use_pool_same]').prop('checked') ? 1 : 0;
                                params.is_limited = $('input[name=is_limited]').prop('checked') ? 1 : 0;
                                contentIds = [];
                                $('.input-content-ids:checked').each(function () {
                                    contentIds.push($(this).val());
                                });
                                if ($(e.target).attr('class_id')) {
                                    params.class_id = $(e.target).attr('class_id');
                                    params.directory_id = $(e.target).attr('directory_id');
                                } else {
                                    params.class_id = 85131;
                                    params.directory_id = $(e.target).attr('directory_id');
                                }
                                params.content_ids = contentIds;
                                if (mode == 'share') {
                                    params.is_share = 1;
                                }
                                glexa.ajax({
                                    action: 'teacher_ajax_class_content_list_copy_accept',
                                    params: params,
                                    onSuccess: function () {
                                        glexa.loadClassDirectories(85131, 1214705);
                                        glexa.loadClassContents(85131, 1214705, 'edit');
                                    }
                                });
                            }
                        });
                        break;
                    case 'backup':
                        $('.div-backup-form').show();
                        $('.td-content-checkbox').show();
                        $('.button-send-backup').on('click', function (e) {
                            e.preventDefault();
                            if (!$('.input-content-ids:checked').length) {
                                glexa.alert('対象の教材にチェックを入れてください');
                                return;
                            }
                            glexa.confirm('時間がかかる場合があります。実行してもよろしいですか？', function (e) {
                                $(this).prop('disabled', true);
                                area = $('[class="area"]:checked').map(function () {
                                    return 'area[]=' + $(this).val()
                                }).get().join('&');
                                location.href = window.urlbase + '?action=teacher_ajax_class_content_list_backup_accept&class_id=85131&' + $.param($('.input-content-ids:checked')) + '&' + $.param($('.checkbox-backups:checked'));
                                $(this).prop('disabled', false);
                            });
                        });
                        break;
                    case 'remove':
                        $('.td-content-checkbox').show();
                        $('.div-remove-form').show();
                        $('.button-send-remove').on('click', function (e) {
                            e.preventDefault();
                            glexa.confirm('本当に削除してもよろしいですか？<br>共有中の教材は、共有先の教材も削除されます。', function (e) {
                                contentIds = [];
                                $('.input-content-ids:checked').each(function () {
                                    contentIds.push($(this).val());
                                });
                                glexa.ajax({
                                    action: 'teacher_ajax_class_content_list_remove_accept',
                                    params: {
                                        class_id: 85131,
                                        directory_id: 1214705,
                                        content_ids: contentIds
                                    },
                                    onSuccess: function () {
                                        glexa.loadClassDirectories(85131, 1214705);
                                        glexa.loadClassContents(85131, 1214705, 'edit');
                                    }
                                });
                            });
                        });
                        break;
                }
            } else {
                glexa.sessionStorage.removeItem('content_edit_mode');
                $('.a-content-menu').css({ opacity: 1 });
            }
        }
    }
    function helperContent() {
        $(this).closest('tr').find('.input-content-ids').prop('checked', true);
        names = [];
        $('.input-content-ids:checked').each(function () {
            names.push($(this).closest('tr').find('.x-drag-title').html());
        });
        return '<fieldset style="min-width:300px;margin-top:30px;"><span class="badge badge-danger">' + names.length + '</span> ' + names.join(',').substr(0, 120) + '</fieldset>';
    }
    function setDirectoryId(directoryId) {
        $('#div-class-contents').attr('directory_id', directoryId);
    }
    function setEnable(contentId) {
        glexa.ajax({
            action: 'teacher_ajax_class_content_list_enable_accept',
            params: {
                content_id: contentId
            },
            onSuccess: function () {
                glexa.loadClassContents(85131, 1214705);
            }
        });
    }
    function saveContentTitle(object) {
        glexa.ajax({
            action: 'teacher_ajax_class_content_list_title_accept',
            params: {
                content_id: object.attr('content_id'),
                title: object.val()
            },
            onSuccess: function () {
                glexa.loadClassContents(85131, 1214705, 'edit');
            }
        });
    }
    function saveContentSort(contentIds) {
        glexa.ajax({
            action: 'teacher_ajax_class_content_list_sort_accept',
            params: {
                class_id: 85131,
                directory_id: 1214705,
                content_ids: contentIds
            },
            onSuccess: function (res) {
                //glexa.loadClassContents(85131, 1214705, 'edit');
            }
        });
    }
    function openModalRestoreForm(classId, directoryId) {
        params = {
            class_id: 85131,
            directory_id: directoryId
        }
        glexa.openRemoteModal({
            action: 'teacher_modal_class_content_restore_form',
            params: params
        });
    }
    function openModalRestoreWebTubeForm(classId, directoryId) {
        params = {
            class_id: 85131,
            directory_id: directoryId
        }
        glexa.openRemoteModal({
            action: 'teacher_modal_class_content_restore_webtube_form',
            params: params
        });
    }
    function getContentList(classId, directoryId, viewType) {
        params = {
            view_type: viewType
        }
        if (classId) {
            params.class_id = classId;
        }
        if (directoryId) {
            params.directory_id = directoryId;
        }
        glexa.ajax({
            action: 'glexa_ajax_class_content_list',
            element: '#div-class-contents',
            params: params
        });
    }
    function editShareContent(pluginKey, pluginId, contentId) {
        glexa.confirm('共有された全ての教材に影響がありますがよろしいですか？', function (e) {
            location.href = '/plugin/' + pluginKey + '/edit/' + pluginId + '/';
        });
    }
    function openModalGrades(contentId) {
        params = {
            class_id: 85131,
            content_id: contentId
        }
        glexa.openRemoteModal({
            action: 'teacher_modal_class_content_grade_list',
            params: params
        });
    }
</script>
<script>
    $(function () {
        // event setup
        $('.a-open-list').on('click', function (e) {
            e.preventDefault();
            openModalDownloadList($(this).attr('file_id'), 85131, $('#div-class-contents').attr('directory_id'));
        });
        // first load
    });
    function openModalDownloadList(fileId, classId, directoryId) {
        params = {
            file_id: fileId,
            class_id: classId,
        }
        if (directoryId) {
            params.directory_id = directoryId;
        }
        glexa.openRemoteModal({
            action: 'teacher_modal_class_resource_file_download_list',
            params: params
        });
    }
</script>
<style>
    .div-small {
        font-size: 80%;
    }

    @media screen and (max-width: 767px) {

        /* sp-resize-table */
        .table-responsive.sp-resize-table table>thead>tr>th,
        .table-responsive.sp-resize-table table>tbody>tr>th,
        .table-responsive.sp-resize-table table>tfoot>tr>th,
        .table-responsive.sp-resize-table table>thead>tr>td,
        .table-responsive.sp-resize-table table>tbody>tr>td,
        .table-responsive.sp-resize-table table>tfoot>tr>td {
            border: solid 1px #dbe4e3;
        }
    }
</style>
<div class="panel panel-default">
    <div class="panel-heading cf">
        <i class="mark"></i>
        第1回 マージと分割統治法
        - 教材一覧
    </div>
    <div class="panel-body sp-padding-none">
        <div class="form-inline">
            <p class="display-inline-block margin-right-lg sp-margin-md">受講したい教材名をクリックしてください</p>
            <select class="form-control select-load-contents sp-margin-md sp-margin-top-none" id="view_type"
                name="view_type" onchange="glexa.getContentList(85131,1214705,'')">
                <option value="1">すべて</option>
                <option value="2">受講済み</option>
                <option value="3">未受講</option>
                <option value="4">受講可能</option>
                <option value="5">期限切れ</option>
                <option value="6">締め切り順</option>
            </select>
            <div class="alert alert-info">
                受講中はWebブラウザの「←(戻る)」「→(進む)」ボタンを使用しないでください。<br>一度に複数のタブやブラウザを開いて受講しないでください。<br>また、同時に複数の端末（PC及びスマホ）から受講しないでください。<br>成績が正しく送信されない可能性があります。
            </div>
        </div><!--form-inline-->
        <div class="table-responsive sp-resize-table border-none">
            <table class="table-class-content">
                <thead>
                    <tr>
                        <th width="40"></th>
                        <th>
                            <div class="div-inner md">教材名</div>
                        </th>
                    </tr>
                </thead>
                <tbody class="tbody-content-sort tbody-content-select">
                    <tr>
                        <td class="border-right-none vertical-align-top">
                            <div class="plugin-icon-wrapper ">
                                <img src="/plugins/board/images/icon_transparent.png" class="plugin-icon" />
                                <i class="state"></i>
                            </div>
                        </td>
                        <td class="border-left-nonevertical-align-middle">
                            <div>
                                <b>
                                    <a href="#" class="a-content-open" content_id="812366" plugin_key="board">
                                        1回目のボード
                                    </a>
                                </b>
                                <div class="div-small">
                                </div>
                            </div>
                        </td>
                    </tr>
                    <tr class="toggle-area x-content-hide">
                        <td colspan="2">
                            <div class="x-content-hide">
                                <div class="description">1回目のボードです。</div>
                                <div class="confirm">
                                    <div class="text-center">
                                        <a href="/content/812366/board/e00329344dde4a39c1cc0fe3bca98d901c87342973f83defe1bd32bfebf8c4fa/"
                                            class="btn btn-black size-sm margin-top-sm margin-bottom-md ">
                                            受講
                                        </a>
                                    </div>
                                </div>
                            </div>
                        </td>
                    </tr>
                    <tr>
                        <td class="border-right-none vertical-align-top">
                            <div class="plugin-icon-wrapper ">
                                <img src="/common/images/icons/resource_icon_transparent.png" class="plugin-icon" />
                                <i class="state"></i>
                            </div>
                        </td>
                        <td class="border-left-none border-right-none" colspan="2">
                            <div>
                                <div class="x-drag-title">
                                    <div class="cf">
                                        <details>
                                            <summary>▼アルゴリズム1 最後の話題をクイズにすると</summary>
                                            クイズ：　ソート問題　　解答例：　ソート問題解答例 そうだったなぁ、と思い出しておいてください<br /><br />
                                        </details>
                                        <details>
                                            <summary>▼パワーポイントについて</summary>
                                            <span>パワーポイントは教科書とセットになっています。</span><br /><span>パワーポイントだけを見ればよい、という意図ではありません。</span><br /><br /><span>パワーポイントは、講義で利用するための資料なので、口語的だったり、文章が不完全だったりします。教科書の表現も確認するようにしてください。<br /><br /></span>
                                        </details>
                                        <details>
                                            <summary>▼教科書のアルゴリズム</summary>
                                            <span>教科書の疑似コードを基に説明します。この分野の伝統的な知識です。疑似コードを参考にせずに自作したとしても、必ず疑似コード（とプログラム）も理解するようにしてください。</span><br /><br /><span>疑似コードを実装しなさい、と指示がある場合、そうでないプログラムはそのままでは減点になります。説明をつけてもらえれば助かります。<br /><br /></span>
                                        </details>
                                        <details>
                                            <summary>▼今日の流れ</summary>
                                            ライブとビデオ（2020年度のもの）で行きます。内容が20年度になっている部分はコメントします。<br /><br />0. イントロ <br />1.
                                            授業イントロ　<br /><br />（アルゴ１を思いだそう）<br />2. 選択ソート<br />3. 配列と再帰<br />4.
                                            練習問題<br /><br />（今日のメインイベント）<br />5. マージ<br />6.
                                            課題確認　(レポートです）（ChatGPTでもなんでも利用してよい。利用する力と自分の理解力をつけること）
                                        </details>
                                    </div>
                                </div>
                                <div class="margin-top-md">
                                    <div class="cf margin-bottom-lg">
                                        <div class="display-block">
                                            <img src="/common/images/icons/pdf.png" alt="" class="icon" /> <a
                                                href="/file/mse3ewdw8/" target="_blank">ソート問題.pdf</a>
                                        </div>
                                    </div>
                                    <div class="cf margin-bottom-lg">
                                        <div class="display-block">
                                            <img src="/common/images/icons/pdf.png" alt="" class="icon" /> <a
                                                href="/file/VpECRNWn/" target="_blank">ソート問題解答例.pdf</a>
                                        </div>
                                    </div>
                                    <div class="cf margin-bottom-lg">
                                        <div class="display-block">
                                            <img src="/common/images/icons/pdf.png" alt="" class="icon" /> <a
                                                href="/file/DJzAc5WWax/" target="_blank">1.pdf</a>
                                        </div>
                                    </div>
                                </div>
                            </div>
                        </td>
                    </tr>
                    <tr>
                        <td class="border-right-none vertical-align-top">
                            <div class="plugin-icon-wrapper img-checked">
                                <img src="/plugins/quiz/images/icon_transparent.png" class="plugin-icon" />
                                <i class="state"></i>
                            </div>
                        </td>
                        <td class="border-left-nonevertical-align-middle">
                            <div>
                                <b>
                                    <a href="#" class="a-content-open" content_id="812368" plugin_key="quiz">
                                        レポートの大枠
                                    </a>
                                </b>
                                <div class="div-small">
                                </div>
                            </div>
                        </td>
                    </tr>
                    <tr class="toggle-area x-content-hide">
                        <td colspan="2">
                            <div class="x-content-hide">
                                <div class="description">
                                    <ul>
                                        <li>（　）は、私からのコメント。作成時には、忘れずに削除しよう。</li>
                                        <li>授業のホームページには、アルゴリズム１の最初のうますぎる例を示してあります。（はじめに、がない頃）</li>
                                        <li>今回は、章が4つです。</li>
                                        <li>1. はじめに の文章例を参考に。次回は、ぜひ自分の言葉で書いてみてください。</li>
                                    </ul>
                                    <hr />
                                    <p>学籍番号 学年 ゼミ 氏名</p>
                                    <ol>
                                        <li>はじめに<br /> （何のレポートなのかを、文章にしよう） （下は例文です）<br />
                                            2つの配列をマージするプログラム(STEP1.1)と実行例を報告する。また時間計算量を分析する。</li>
                                        <li>STEP1 マージのプログラム　<br /> （コメントをがんばろう。アルゴリズムの説明です）</li>
                                        <li>実行例<br /> (STEP1.1の例とは異なるものも示すとよいです）</li>
                                        <li>時間計算量<br /> 　（「O ( ) です」。だけではなくて、説明をしよう。教科書を参考にしよう）</li>
                                        <li>おわりに（感想、コメントなどあれば）</li>
                                    </ol>
                                </div>
                                <div class="confirm">
                                    <div class="text-center">
                                        <a href="/content/812368/quiz/29d342429111eedda13f712239c4f4cde1d7db4f4fde1b234a60e0c559811dba/"
                                            class="btn btn-black size-sm margin-top-sm margin-bottom-md ">
                                            受講
                                        </a>
                                        <a href="/plugin/quiz/result/151540/"
                                            class="btn btn-black size-sm margin-top-sm margin-bottom-md">
                                            結果
                                        </a>
                                    </div>
                                </div>
                            </div>
                        </td>
                    </tr>
                    <tr>
                        <td class="border-right-none vertical-align-top">
                            <div class="plugin-icon-wrapper img-checked">
                                <img src="/plugins/report/images/icon_transparent.png" class="plugin-icon" />
                                <i class="state"></i>
                            </div>
                        </td>
                        <td class="border-left-nonevertical-align-middle">
                            <div>
                                <b>
                                    <a href="#" class="a-content-open" content_id="812371" plugin_key="report">
                                        課題レポート
                                    </a>
                                </b>
                                <div class="div-small">
                                    <b>提出受付期間:</b> ～ 4月14日(月) 18:05
                                </div>
                            </div>
                        </td>
                    </tr>
                    <tr class="toggle-area x-content-hide">
                        <td colspan="2">
                            <div class="x-content-hide">
                                <div class="description">
                                    <p>〆切は、月曜日18:00です。<br />(ファイル名は半数英数字、テキストファイルをアップロードしてください）<br /><br /><span>アルゴリズム２は、細かくて面倒な、ミスすると動かないプログラムになっていきます。</span><br /><span>うっかりミスに、イライラするときもあるかもしれませんが、がんばってください。ミスやエラーを何とかしていく力も重要。<br /><br /></span>
                                    </p>
                                </div>
                                <div class="confirm">
                                    <div class="text-center">
                                        <a href="/content/812371/report/57a2f39c7630a0b277b3f17dc8fdc59995c22e8b0fe147b79c3322e089dcd3b9/"
                                            class="btn btn-black size-sm margin-top-sm margin-bottom-md ">
                                            受講
                                        </a>
                                        <a href="/plugin/report/result/136338/"
                                            class="btn btn-black size-sm margin-top-sm margin-bottom-md">
                                            結果
                                        </a>
                                    </div>
                                </div>
                            </div>
                        </td>
                    </tr>
                    <tr>
                        <td class="border-right-none vertical-align-top">
                            <div class="plugin-icon-wrapper img-checked">
                                <img src="/plugins/quiz/images/icon_transparent.png" class="plugin-icon" />
                                <i class="state"></i>
                            </div>
                        </td>
                        <td class="border-left-nonevertical-align-middle">
                            <div>
                                <b>
                                    <a href="#" class="a-content-open" content_id="812369" plugin_key="quiz">
                                        参考資料について
                                    </a>
                                </b>
                                <div class="div-small">
                                </div>
                            </div>
                        </td>
                    </tr>
                    <tr class="toggle-area x-content-hide">
                        <td colspan="2">
                            <div class="x-content-hide">
                                <div class="description">参考資料<br />
                                    <ul>
                                        <li>１年生春の「コンピュータのアーキテクチャと構造」の教科書
                                            <ul>
                                                <li>10章：来週とりあげるマージソートと　アルゴリズム１の最終回に学んだバブルソートを例にしています。マージソートは、再帰を扱うので、再帰についての紹介もあります。
                                                </li>
                                                <li>9章：C言語とコンパイル過程、実行過程の説明になっています。</li>
                                            </ul>
                                        </li>
                                        <li>長谷川先生のPythonの教科書 (2023年度生)「Python言語によるプログラムイントロダクション」
                                            <ul>
                                                <li>p.151 からソート。マージソートも紹介があります。</li>
                                                <li>比較演算子も変数（引数）にしているところが、この教科書らしい。</li>
                                            </ul>
                                        </li>
                                    </ul>
                                </div>
                                <div class="confirm">
                                    <div class="text-center">
                                        <a href="/content/812369/quiz/06a1f270fbe93e0477869f7d44fa2ee1f3c83502ab8a1161a413c6186c5df738/"
                                            class="btn btn-black size-sm margin-top-sm margin-bottom-md ">
                                            受講
                                        </a>
                                        <a href="/plugin/quiz/result/151541/"
                                            class="btn btn-black size-sm margin-top-sm margin-bottom-md">
                                            結果
                                        </a>
                                    </div>
                                </div>
                            </div>
                        </td>
                    </tr>
                    <tr>
                        <td class="border-right-none vertical-align-top">
                            <div class="plugin-icon-wrapper ">
                                <img src="/plugins/quiz/images/icon_transparent.png" class="plugin-icon" />
                                <i class="state"></i>
                            </div>
                        </td>
                        <td class="border-left-nonevertical-align-middle">
                            <div>
                                <b>
                                    <a href="#" class="a-content-open" content_id="812370" plugin_key="quiz">
                                        今日のビデオ
                                    </a>
                                </b>
                                <div class="div-small">
                                </div>
                            </div>
                        </td>
                    </tr>
                    <tr class="toggle-area x-content-hide">
                        <td colspan="2">
                            <div class="x-content-hide">
                                <div class="description">
                                    <div>2022.4.7
                                        1時間目の授業はMaNaBoにトラブルが発生してしまったために、授業の内容にアクセスできなかった人がいたかもしれません。授業は、下のビデオに2022年度向けコメントを追加する形で進みました。<br /><br />
                                    </div>
                                    <ul>
                                        <li>ライブとビデオ（2020年度のもの）で行きます。</li>
                                        <li>内容が20年度になっている部分はコメントします。</li>
                                        <li>ビデオ音は、標準化しました。<br /><br /></li>
                                        <li>イントロ
                                            <ul>
                                                <li>0. イントロ <br />https://youtu.be/CgZ2vS6IuDw</li>
                                                <li>1. 授業イントロ　<br />https://youtu.be/xbYcwLwh6sU</li>
                                            </ul>
                                        </li>
                                        <li>アルゴ１を思いだそう
                                            <ul>
                                                <li>2. 選択ソート<br />https://youtu.be/CDJ7PajWKbU</li>
                                                <li>3. 配列と再帰<br />https://youtu.be/pxo_MM2djXg</li>
                                                <li>4. 練習問題<br />https://youtu.be/RQMrbSqj4Ps<br /><br /></li>
                                            </ul>
                                        </li>
                                        <li>今日のメインイベント
                                            <ul>
                                                <li>5. マージ<br />https://youtu.be/2DU6kNAY08E</li>
                                                <li>6. 課題確認　(レポートです）<br />https://youtu.be/iJEH-zgZcWM<br /><br /></li>
                                            </ul>
                                        </li>
                                    </ul>
                                    <p>@m.chukyo-u.ac.jp のアカウントのみの限定公開になっています。</p>
                                </div>
                                <div class="confirm">
                                    <div class="text-center">
                                        <a href="/content/812370/quiz/2cb92c2ceb1b5f53d5e62824079420d9035200aec0276f0a3377794b5382cc90/"
                                            class="btn btn-black size-sm margin-top-sm margin-bottom-md ">
                                            受講
                                        </a>
                                    </div>
                                </div>
                            </div>
                        </td>
                    </tr>
                </tbody>
            </table>
            <div class="x-content-hide div-backup-form margin-top-lg">
                <div class="alert alert-default text-right">
                    <button class="button-send-backup btn btn-primary btn-sm">実行</button>
                </div>
            </div>
            <div class="x-content-hide div-remove-form margin-top-lg">
                <div class="alert alert-default text-right">
                    <button class="button-send-remove btn btn-danger btn-sm">実行</button>
                </div>
            </div>
        </div><!--table-responsive-->
    </div><!--panel-body-->
</div><!--panel-->
