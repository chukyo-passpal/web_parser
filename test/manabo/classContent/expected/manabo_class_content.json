{
  "items": [
    {
      "contentId": "812366",
      "pluginKey": "board",
      "title": "1回目のボード",
      "iconSrc": "/plugins/board/images/icon_transparent.png",
      "descriptionHtml": "1回目のボードです。",
      "action": {
        "label": "受講",
        "url": "/content/812366/board/e00329344dde4a39c1cc0fe3bca98d901c87342973f83defe1bd32bfebf8c4fa/"
      }
    },
    {
      "contentId": "812368",
      "pluginKey": "quiz",
      "title": "レポートの大枠",
      "iconSrc": "/plugins/quiz/images/icon_transparent.png",
      "descriptionHtml": "<ul>\n                                        <li>（　）は、私からのコメント。作成時には、忘れずに削除しよう。</li>\n                                        <li>授業のホームページには、アルゴリズム１の最初のうますぎる例を示してあります。（はじめに、がない頃）</li>\n                                        <li>今回は、章が4つです。</li>\n                                        <li>1. はじめに の文章例を参考に。次回は、ぜひ自分の言葉で書いてみてください。</li>\n                                    </ul>\n                                    <hr>\n                                    <p>学籍番号 学年 ゼミ 氏名</p>\n                                    <ol>\n                                        <li>はじめに<br> （何のレポートなのかを、文章にしよう） （下は例文です）<br>\n                                            2つの配列をマージするプログラム(STEP1.1)と実行例を報告する。また時間計算量を分析する。</li>\n                                        <li>STEP1 マージのプログラム　<br> （コメントをがんばろう。アルゴリズムの説明です）</li>\n                                        <li>実行例<br> (STEP1.1の例とは異なるものも示すとよいです）</li>\n                                        <li>時間計算量<br> 　（「O ( ) です」。だけではなくて、説明をしよう。教科書を参考にしよう）</li>\n                                        <li>おわりに（感想、コメントなどあれば）</li>\n                                    </ol>",
      "action": {
        "label": "受講",
        "url": "/content/812368/quiz/29d342429111eedda13f712239c4f4cde1d7db4f4fde1b234a60e0c559811dba/"
      }
    },
    {
      "contentId": "812371",
      "pluginKey": "report",
      "title": "課題レポート",
      "iconSrc": "/plugins/report/images/icon_transparent.png",
      "descriptionHtml": "<p>〆切は、月曜日18:00です。<br>(ファイル名は半数英数字、テキストファイルをアップロードしてください）<br><br><span>アルゴリズム２は、細かくて面倒な、ミスすると動かないプログラムになっていきます。</span><br><span>うっかりミスに、イライラするときもあるかもしれませんが、がんばってください。ミスやエラーを何とかしていく力も重要。<br><br></span>\n                                    </p>",
      "action": {
        "label": "受講",
        "url": "/content/812371/report/57a2f39c7630a0b277b3f17dc8fdc59995c22e8b0fe147b79c3322e089dcd3b9/"
      }
    },
    {
      "contentId": "812369",
      "pluginKey": "quiz",
      "title": "参考資料について",
      "iconSrc": "/plugins/quiz/images/icon_transparent.png",
      "descriptionHtml": "参考資料<br>\n                                    <ul>\n                                        <li>１年生春の「コンピュータのアーキテクチャと構造」の教科書\n                                            <ul>\n                                                <li>10章：来週とりあげるマージソートと　アルゴリズム１の最終回に学んだバブルソートを例にしています。マージソートは、再帰を扱うので、再帰についての紹介もあります。\n                                                </li>\n                                                <li>9章：C言語とコンパイル過程、実行過程の説明になっています。</li>\n                                            </ul>\n                                        </li>\n                                        <li>長谷川先生のPythonの教科書 (2023年度生)「Python言語によるプログラムイントロダクション」\n                                            <ul>\n                                                <li>p.151 からソート。マージソートも紹介があります。</li>\n                                                <li>比較演算子も変数（引数）にしているところが、この教科書らしい。</li>\n                                            </ul>\n                                        </li>\n                                    </ul>",
      "action": {
        "label": "受講",
        "url": "/content/812369/quiz/06a1f270fbe93e0477869f7d44fa2ee1f3c83502ab8a1161a413c6186c5df738/"
      }
    },
    {
      "contentId": "812370",
      "pluginKey": "quiz",
      "title": "今日のビデオ",
      "iconSrc": "/plugins/quiz/images/icon_transparent.png",
      "descriptionHtml": "<div>2022.4.7\n                                        1時間目の授業はMaNaBoにトラブルが発生してしまったために、授業の内容にアクセスできなかった人がいたかもしれません。授業は、下のビデオに2022年度向けコメントを追加する形で進みました。<br><br>\n                                    </div>\n                                    <ul>\n                                        <li>ライブとビデオ（2020年度のもの）で行きます。</li>\n                                        <li>内容が20年度になっている部分はコメントします。</li>\n                                        <li>ビデオ音は、標準化しました。<br><br></li>\n                                        <li>イントロ\n                                            <ul>\n                                                <li>0. イントロ <br>https://youtu.be/CgZ2vS6IuDw</li>\n                                                <li>1. 授業イントロ　<br>https://youtu.be/xbYcwLwh6sU</li>\n                                            </ul>\n                                        </li>\n                                        <li>アルゴ１を思いだそう\n                                            <ul>\n                                                <li>2. 選択ソート<br>https://youtu.be/CDJ7PajWKbU</li>\n                                                <li>3. 配列と再帰<br>https://youtu.be/pxo_MM2djXg</li>\n                                                <li>4. 練習問題<br>https://youtu.be/RQMrbSqj4Ps<br><br></li>\n                                            </ul>\n                                        </li>\n                                        <li>今日のメインイベント\n                                            <ul>\n                                                <li>5. マージ<br>https://youtu.be/2DU6kNAY08E</li>\n                                                <li>6. 課題確認　(レポートです）<br>https://youtu.be/iJEH-zgZcWM<br><br></li>\n                                            </ul>\n                                        </li>\n                                    </ul>\n                                    <p>@m.chukyo-u.ac.jp のアカウントのみの限定公開になっています。</p>",
      "action": {
        "label": "受講",
        "url": "/content/812370/quiz/2cb92c2ceb1b5f53d5e62824079420d9035200aec0276f0a3377794b5382cc90/"
      }
    }
  ]
}